src/screens/EditorScreen.tsx

```diff
@@ -125,10 +127,11 @@ export function EditorScreen() {
             setLastChange(prevLastChanges => currentLastChanges.length > 0 ? currentLastChanges : prevLastChanges);
         }
         setCurrentContent(content);
-        setDirty(true);
-    }, [currentContent]);
+        setDirty(hash(content) !== fileHash);
+    }, [fileHash, currentContent]);
     const loadNewDoc = useCallback((content: Content | string) => {
         setDirty(false);
+        setFileHash(hash(content));
```

Comment:
I think we also need to setFileHash on page save.

Reply:
Both `doSave` and `doSaveAs` call `context.editor.loadNewDoc`, so I'm pretty sure this already happens!